File: swiftui/Sources/ArenaTokens/ArenaControl.swift
import CoreGraphics
import SwiftUI

public enum ArenaControl {
    public static let padding: CGFloat = ArenaTokens.sp3

    public static let gap: CGFloat = ArenaTokens.sp2

    public static let radius: CGFloat = ArenaTokens.rSm

    public static let weight: Font.Weight = ArenaTokens.fwSemibold

    public static func height(_ size: ArenaControlSize, _ density: ArenaDensityScale) -> CGFloat {
        switch size {
        case .sm: density.ctlHSm
        case .md: density.ctlH
        case .lg: density.ctlHLg
        }
    }

    public static func text(_ size: ArenaControlSize, _ density: ArenaDensityScale) -> CGFloat {
        switch size {
        case .sm: density.textMd
        case .md: density.text
        case .lg: density.text
        }
    }

    public static func target(_ painted: CGFloat, floor: CGFloat) -> CGFloat { max(floor, painted) }
}

File: compose/src/main/kotlin/org/dravensoft/arena/tokens/ArenaControl.kt
package org.dravensoft.arena.tokens

import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.max
import org.dravensoft.arena.api.ArenaControlSize

public object ArenaControl {
    public val padding: Dp = ArenaTokens.sp3

    public val gap: Dp = ArenaTokens.sp2

    public val radius: Dp = ArenaTokens.rSm

    public val weight: FontWeight = ArenaTokens.fwSemibold

    public fun height(size: ArenaControlSize, density: ArenaDensityScale): Dp = when (size) {
        ArenaControlSize.Sm -> density.ctlHSm
        ArenaControlSize.Md -> density.ctlH
        ArenaControlSize.Lg -> density.ctlHLg
    }

    public fun text(size: ArenaControlSize, density: ArenaDensityScale): TextUnit = when (size) {
        ArenaControlSize.Sm -> density.textMd
        ArenaControlSize.Md -> density.text
        ArenaControlSize.Lg -> density.text
    }

    public fun target(painted: Dp, floor: Dp): Dp = max(floor, painted)
}
